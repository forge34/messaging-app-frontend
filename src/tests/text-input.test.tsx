import { render, screen } from "@testing-library/react";
import TextInput from "../components/text-input";
import userEvent from "@testing-library/user-event";

describe("text input test suite", () => {
  test("should match snapshot", () => {
    const { container } = render(
      <TextInput name="username" label="Username"></TextInput>,
    );

    expect(container).toMatchSnapshot();
  });
  test("should render label correctly", () => {
    render(<TextInput name="username" label="Username"></TextInput>);

    expect(screen.getByText("Username".toUpperCase())).toBeInTheDocument();
  });

  test("should render input", () => {
    render(<TextInput name="username" label="Username"></TextInput>);

    expect(screen.getByRole("textbox")).toBeInTheDocument();
  });

  test("should change input value on typing", async () => {
    const user = userEvent.setup();
    render(<TextInput name="username" label="Username"></TextInput>);

    const input = screen.getByRole<HTMLInputElement>("textbox");
    await user.type(input, "Forge");

    expect(input.value).toMatch("Forge");
  });

  test("should focus input on click", async () => {
    const user = userEvent.setup();
    render(<TextInput name="username" label="Username"></TextInput>);

    const input = screen.getByRole<HTMLInputElement>("textbox");
    await user.click(input);

    expect(input).toHaveFocus();
  });
});
