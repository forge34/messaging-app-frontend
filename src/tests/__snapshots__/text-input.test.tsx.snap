// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`text input test suite > should match snapshot 1`] = `
<div>
  <div
    class="text-input"
  >
    <div>
      <label>
        USERNAME
         
      </label>
    </div>
    <input
      class=""
      name="username"
      type="text"
      value=""
    />
  </div>
</div>
`;
